### PR TITLE
[W4.3d][W13-A2]Yang Minxing

### PR DESCRIPTION
**Pull request by yangminxingnus**
Add class block street unit postalcode.

### CODE REVIEW

File: src/seedu/addressbook/commands/CommandResult.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class CommandResult {
 
     /** The feedback message to be shown to the user. Contains a description of the execution result */
-    public final String feedbackToUser;
+    private final String feedbackToUser;
```
Comment (hovinh) [inline src/seedu/addressbook/commands/CommandResult.java]: 👍 for encapsulation. The commit message is quite long, you can find another way to express it. Check the regulation when making a commit message.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 2]
```diff
@@ -92,7 +95,7 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
```
Comment (hovinh) [inline src/seedu/addressbook/storage/StorageFile.java]: Update the comment

File: src/seedu/addressbook/ui/Formatter.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Scanner;
+
+import static seedu.addressbook.common.Messages.*;
```
Comment (hovinh) [inline src/seedu/addressbook/ui/Formatter.java]: Can you tell me why one should not use * here?

File: src/seedu/addressbook/ui/Formatter.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import static seedu.addressbook.common.Messages.MESSAGE_INIT_FAILED;
+
+public class Formatter{
```
Comment (hovinh) [inline src/seedu/addressbook/ui/Formatter.java]: Header comment should be here.

File: src/seedu/addressbook/ui/Formatter.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    /** Offset required to convert between 1-indexing and 0-indexing.  */
+    public static final int DISPLAYED_INDEX_OFFSET = 1;
+
+    /** Format of a comment input line. Comment lines are silently consumed when reading user input. */
+    private static final String COMMENT_LINE_FORMAT_REGEX = "#.*";
+
+
+    private final PrintStream out;
+
+    public Formatter() {
```
Comment (hovinh) [inline src/seedu/addressbook/ui/Formatter.java]: It's nice to see you follow the rule of order of declaring methods 👍

File: src/seedu/addressbook/data/person/Block.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package seedu.addressbook.data.person;
+
```
Comment (hovinh) [inline src/seedu/addressbook/data/person/Block.java]: You should have header comment here to explain this class.
Moreover, each commit should be corresponding to one single change, which is for one class only.

File: src/seedu/addressbook/data/person/Address.java [hunk 7]
```diff
@@ -2,15 +2,25 @@
...
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in format of a/BLOCK, STREET, UNIT, POSTAL_CODE ";
```
Comment (hovinh) [inline src/seedu/addressbook/data/person/Address.java]: 👍  for updating the IllegalValueException

Comment (hovinh) [reply src/seedu/addressbook/data/person/Address.java]: This line is a quite long.

File: src/seedu/addressbook/commands/AddCommand.java [hunk 8]
```diff
@@ -23,9 +23,9 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
             + "Contact details can be marked private by prepending 'p' to the prefix.\n"
-            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS(a/BLOCK, STREET, UNIT, POSTAL_CODE)  [t/TAG]...\n"
```
Comment (hovinh) [inline src/seedu/addressbook/commands/AddCommand.java]: 👍  for clear instruction. Nice to see you update the help function.

File: src/seedu/addressbook/data/person/Address.java [hunk 9]
```diff
@@ -29,11 +39,26 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.value = trimmedAddress;
     }
 
+    private void seperateAddress(String address){
+        String[] eachInfo = address.split(",");
+        block = new Block(eachInfo[0].trim());
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: What does our coding standard say about using numbers in array indices? [Hint! read magic numbers]

File: src/seedu/addressbook/data/person/Block.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+        value = block;
+    }
+
+    public String getBlock() {
+        return value;
+    }
+
+    public boolean isValid() {
+        for(int i = 0; i < value.length(); i++) {
+            if(!Character.isLetterOrDigit(value.charAt(i))) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: you can try to use regular expressions here.

File: src/seedu/addressbook/commands/CommandResult.java (hunk 1 above)
Comment (okkhoy) [inline src/seedu/addressbook/commands/CommandResult.java]: additionally, you need to update the files that use `feedbackToUser` member.

**Comment by okkhoy** [comment]
@yangminxingnus 
Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**